src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/rubygems/GemlockNodeParser.groovy

```diff
@@ -49,32 +49,31 @@ class GemlockNodeParser {
     private boolean inSpecsSection = false
     private boolean inDependenciesSection = false
 
-    DependencyGraph parseProjectDependencies(NameVersionNodeTransformer nameVersionNodeTransformer, final String gemfileLockContents) {
+    DependencyGraph parseProjectDependencies(NameVersionNodeTransformer nameVersionNodeTransformer, final List<String> gemfileLockLines) {
         rootNameVersionNode = new NameVersionNodeImpl([name: 'gemfileLockRoot'])
         nameVersionNodeBuilder = new NameVersionNodeBuilder(rootNameVersionNode)
         directDependencyNames = new HashSet<>()
         currentParent = null
 
-        String[] lines = gemfileLockContents.split(System.lineSeparator())
-        for (String line : lines) {
+        gemfileLockLines.each{ String line ->
             if (!line?.trim()) {
```

Comment:
Why dont you trim the line and use that, rather than trimming it here and again on line 65 and 70?